python/ray/projects/projects.py

```diff
@@ -0,0 +1,103 @@
+import json
+import jsonschema
+import os
+import yaml
+
+
+def find_root(dir):
+    """Find root directory of the ray project.
+
+    Args:
+        dir (str): Directory to start the search in.
+
+    Returns:
+        Path of the parent directory containing the .rayproject or
+        None if no such project is found.
+    """
+    prev, dir = None, os.path.abspath(dir)
+    while prev != dir:
+        if os.path.isdir(os.path.join(dir, ".rayproject")):
+            return dir
+        prev, dir = dir, os.path.abspath(os.path.join(dir, os.pardir))
+    return None
+
+
+def validate_project_schema(project_definition):
+    """Validate a project file against the official ray project schema.
+
+    Raises an exception if the project file is not valid.
```

Comment:
In other parts of the Python code, we use a separate block for `Raises` and list the types of exceptions that can be raised ([example](https://github.com/ray-project/ray/blob/master/python/ray/worker.py#L261)).

Reply:
fixed